lib/action_subscriber/publisher/async/in_memory_adapter.rb

```diff
@@ -62,6 +62,7 @@ class AsyncQueue
 
           def initialize
             @queue = ::Queue.new
+            @instrumentation_threadpool = ::Lifeguard::Threadpool.new
```

Comment:
@mmmries I used a threadpool because I don't like the idea of external code degrading the reliability of the supervisor, whether it be slowing the interval down or an exception coming back up through the instrument. I originally wrote this without one, but added this because of my previously stated concerns. I'm happy to rip it out if you think it's safe to do so.